projects/rocprofiler-systems/tests/rocprof-sys-openmp-tests.cmake

```
@@ -52,71 +52,107 @@ endif()
+  REWRITE_RUN_PASS_REGEX "${_OMPT_PASS_REGEX}"
+  REWRITE_FAIL_REGEX "0 instrumented loops in procedure"
 )
 
 rocprofiler_systems_add_test(
-    SKIP_RUNTIME SKIP_REWRITE
-    NAME openmp-target
-    TARGET openmp-target
-    GPU ON
-    LABELS "openmp;openmp-target"
-    ENVIRONMENT
-        "${_ompt_environment};${_rocm_ld_env};ROCPROFSYS_ROCM_DOMAINS=hip_runtime_api,kernel_dispatch"
+  SKIP_RUNTIME SKIP_REWRITE
+  NAME openmp-target
+  TARGET openmp-target
+  GPU ON
+  LABELS "openmp;openmp-target"
+  ENVIRONMENT
+    "${_ompt_environment};${_rocm_ld_env};ROCPROFSYS_ROCM_DOMAINS=hip_runtime_api,kernel_dispatch"
 )
 
 rocprofiler_systems_add_validation_test(
-    NAME openmp-target-sampling
-    PERFETTO_METRIC "rocm_kernel_dispatch"
-    PERFETTO_FILE "perfetto-trace.proto"
-    LABELS "openmp;openmp-target"
-    ENVIRONMENT "${_rocm_ld_env}"
-    ARGS --label-substrings
-         Z4vmulIiEvPT_S1_S1_i_l51.kd
-         Z4vmulIfEvPT_S1_S1_i_l51.kd
-         Z4vmulIdEvPT_S1_S1_i_l51.kd
-         -c
-         4
-         4
-         4
-         -d
-         0
-         0
-         0
-         -p
+  NAME openmp-target-sampling
+  PERFETTO_METRIC "rocm_kernel_dispatch"
+  PERFETTO_FILE "perfetto-trace.proto"
+  LABELS "openmp;openmp-target"
+  ENVIRONMENT "${_rocm_ld_env}"
+  ARGS
+    --label-substrings
+    Z4vmulIiEvPT_S1_S1_i_l51.kd
+    Z4vmulIfEvPT_S1_S1_i_l51.kd
+    Z4vmulIdEvPT_S1_S1_i_l51.kd
+    -c 4 4 4
+    -d 0 0 0
+    -p
 )
 
+# OpenMP tests generated using OMPVV binaries
+if(ROCPROFSYS_OMPVV_HOST_TESTS)
+    foreach(HOST_TEST_NAME ${ROCPROFSYS_OMPVV_HOST_TESTS})
+        rocprofiler_systems_add_test(
+          SKIP_RUNTIME
+          NAME ${HOST_TEST_NAME}
+          TARGET ${HOST_TEST_NAME}
+          LABELS "openmp"
+          REWRITE_ARGS
+            -e -v 2 --instrument-loops
+          RUNTIME_ARGS
+            -e -v 1 --label return args -E ^GOMP
+          REWRITE_TIMEOUT 180
+          RUNTIME_TIMEOUT 360
+          ENVIRONMENT
+            "${_ompt_environment};ROCPROFSYS_USE_SAMPLING=ON;ROCPROFSYS_SAMPLING_FREQ=50;ROCPROFSYS_COUT_OUTPUT=ON"
+          REWRITE_RUN_PASS_REGEX "${_OMPT_PASS_REGEX}"
+          REWRITE_FAIL_REGEX "0 instrumented loops in procedure"
+        )
+    endforeach()
+endif()
+
+if(ROCPROFSYS_OMPVV_OFFLOAD_TESTS)
+    foreach(OFFLOAD_TEST_NAME ${ROCPROFSYS_OMPVV_OFFLOAD_TESTS})
+        rocprofiler_systems_add_test(
+          SKIP_RUNTIME
+          NAME ${OFFLOAD_TEST_NAME}
+          TARGET ${OFFLOAD_TEST_NAME}
+          GPU ON
+          LABELS "openmp"
+          REWRITE_ARGS -e -v 2
+          ENVIRONMENT
+            "${_ompt_environment};${_rocm_ld_env};ROCPROFSYS_USE_SAMPLING=ON;ROCPROFSYS_SAMPLING_FREQ=50;ROCPROFSYS_COUT_OUTPUT=ON;ROCPROFSYS_ROCM_DOMAINS=hip_runtime_api,marker_api,kernel_dispatch,memory_copy,scratch_memory,hsa_api"
+          REWRITE_RUN_PASS_REGEX
+            "${_OMPVV_OFFLOAD_PASS_REGEX}"
+        )
```

```suggestion
    set(_ompvv_offload_environ
        "${_ompt_environment}"
        "${_rocm_ld_env}"
        "ROCPROFSYS_USE_SAMPLING=ON"
        "ROCPROFSYS_SAMPLING_FREQ=50"
        "ROCPROFSYS_COUT_OUTPUT=ON"
        "ROCPROFSYS_ROCM_DOMAINS=hip_runtime_api,marker_api,kernel_dispatch,memory_copy,scratch_memory,hsa_api"
    )

    foreach(OFFLOAD_TEST_NAME ${ROCPROFSYS_OMPVV_OFFLOAD_TESTS})
        rocprofiler_systems_add_test(
            SKIP_RUNTIME
            NAME ${OFFLOAD_TEST_NAME}
            TARGET ${OFFLOAD_TEST_NAME}
            GPU ON
            LABELS "openmp;ompvv;openmp-target"
            REWRITE_ARGS -e -v 2
            ENVIRONMENT
                "${_ompvv_offload_environ}"
            REWRITE_RUN_PASS_REGEX
                "${_OMPVV_OFFLOAD_PASS_REGEX}"
        )
```
Small refactor for readability. Saving the environment to a variable and passing that into the function.